clang/include/clang/Driver/Options.td

```diff
@@ -918,6 +918,9 @@ def Wdeprecated : Flag<["-"], "Wdeprecated">, Group<W_Group>,
   HelpText<"Enable warnings for deprecated constructs and define __DEPRECATED">;
 def Wno_deprecated : Flag<["-"], "Wno-deprecated">, Group<W_Group>,
   Visibility<[ClangOption, CC1Option]>;
+def Wformat_signedness : Flag<["-"], "Wformat-signedness">,
```

Comment:
I think the warning option config should go to `clang/include/clang/Basic/DiagnosticSemaKinds.td` instead of the toplevel driver setting. For example,  you wouldn't have `-Wno-format-signedness` atm.

Reply:
I have tried to mimic the gcc behavior when implementing this and in gcc it seems like -Wformat-signedness don't introduce a new warning but only modify the behavior of the already existing -Wformat warning.
If I place the warning option config in clang/include/clang/Basic/DiagnosticSemaKinds.td as you suggest don't I then also create a new warning with a new warning text? ... that is not what gcc do (as far as I understand).

Well, with that said we don't have to follow exactly how gcc implemented this if we think there is a better way to implement this in clang.

I agree that support for -Wno-format-signednes is missing from my patch and that should be added.

Reply:
There's a lot of good, free integration that you get with a real warning flag:

* -Wno symmetry (as @hazohelet already said)
* control via `#pragma clang diagnostic`
* diagnostic text shows the name of the warning flag (in case you need to look it up or disable it)

-Wformat-pedantic is another warning that "modifies" -Wformat and it exists as its own warning. I think that it would make sense to do the same thing with -Wformat-signedness.

Reply:
I agree with @hazohelet

Reply:
Yup, this will have to change to adding the diagnostic to DiagnosticSemaKinds.td. I think we will want to control this separately from `-Wformat` but have it grouped under that flag.